tests/integration/mem_leak_test.py

```diff
@@ -0,0 +1,27 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import libnmstate
+import resource
+
+
+def test_repeat_call_of_show():
+    resource.setrlimit(resource.RLIMIT_NOFILE, [1000, 2000])
+    for _ in range(0, resource.getrlimit(resource.RLIMIT_NOFILE)[0] + 1):
+        libnmstate.show()
```

Comment:
Since this should check for a file descriptor leak, how about getting the value of `len(os.listdir("/proc/self/fd"))` before and after the loop and asserting that it did not grow (too much)?

Reply:
Good point.
Actually, the opened fd should be remain the same before and after the `libnmstate.show()`.

Coding on it.